src/util.rs

```diff
@@ -222,3 +216,11 @@ pub fn get_file_contents(filename: &str, ctx: &Context) -> Option<Rope> {
                 .and_then(|f| Rope::from_reader(BufReader::new(f)).ok())
         })
 }
+
+pub fn short_file_path<'a>(target: &'a str, current_dir: &str) -> &'a str {
+    Path::new(target)
+        .strip_prefix(current_dir)
```

Comment:
I wonder if we can make the paths relative to Kakoune's directory, instead of the project root.
Otherwise if I edit `src/main.rs`, use `:cd src`, and then call `:lsp-references`, the `src/main.rs` path is invalid. (also that stopped working at all after the `cd`).

I think we want the behavior of `realpath src/main.rs --relative-to $PWD`, where $PWD is the Kakoune server's working directory.

Reply:
This would be a very dirty implementation, by simply overriding the old commands:

```
define-command -override -hidden lsp-show-goto-choices -params 2 -docstring "Render goto choices" %{
	evaluate-commands -save-regs '"' -try-client %opt[toolsclient] %{
		edit! -scratch *goto*
		set-option buffer filetype grep
		set-option buffer grep_current_line 0
		set-register '"' %sh{
			nl='
			'
			IFS=$nl
			printf %s "$2" | while read line; do
			file=${line%%:*}
			rest=${line#*:}
			# Relative file paths are relative to the project root.
			[ x"${file#/}" = x"${file}" ] && file=$1/$file
			printf '%s:%s\n' "$(realpath --relative-to "$PWD" "$file")" "$rest"
			done
		}
		execute-keys Pgg
	}
}
```

I'm not yet sure how this should be implemented